src/hd-account.js

```diff
@@ -211,24 +206,12 @@ HDAccount.factory = function (o) {
 // JSON SERIALIZER
 
 HDAccount.prototype.toJSON = function () {
-  let labelsJSON = this._address_labels_backup;
-
-  // Hold on to the backup until labels are saved in KV store.
-  if (MyWallet.wallet.labels && !MyWallet.wallet.labels.readOnly && !MyWallet.wallet.labels.dirty) {
-    // Use placeholder entry to prevent address reuse:
-    labelsJSON = [];
-    let max = MyWallet.wallet.labels.maxLabeledReceiveIndex(this._index);
-    if (max > -1) {
-      labelsJSON.push({index: max, label: ''});
-    }
-  }
-
   var hdaccount = {
     label: this._label,
     archived: this._archived,
     xpriv: this._xpriv,
     xpub: this._xpub,
-    address_labels: labelsJSON,
+    address_labels: this._address_labels.sort(o => o.index),
```

Comment:
I don't think this will sort correctly, see [MDN sort](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort). The comparison function should return a negative or positive number based on how it should be sorted, like `(a, b) => a - b`.